site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,121 @@
+---
+slug: optimize-your-s3-endpoints
+title: Optimize your S3 endpoints for cost and efficiency
+authors: [yrobla]
+---
+
+# Optimize your S3 endpoints for cost and efficiency
+
+Are you using S3 buckets as part of your cloud deployments ? How are you accessing them?
+
+When not configured properly, access to the S3 buckets may need internet access. One simple but costly way
+is to rely on <a href="https://docs.aws.amazon.com/en_en/vpc/latest/userguide/vpc-nat-gateway.html" target="_blank">NAT gateways</a>.
+
+But there is a more optimal solution, that is creating <a
+href="https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html"
+target="_blank"> gateway or interface VPC endpoints</a> for each region where you have buckets exposed.
+
+When the VPC endpoints are enabled you could access your S3 buckets using this endpoint. This will allow you
+to control your buckets from internal network with the desired security, and without the extra costs of a
+NAT gateway.
+
+### Why use VPC Interface Endpoints
+
+The two main reasons are **security** and **performance**.
+
+**Security: no internet traversal**
+
+When using endpoints, services will be accessed via Amazon's internal networks to route the traffic. This is
+essential for security services or when hosting any content with sensitive data.
+
+**Security: policies**
+
+Many of the servcies are supporting the use of policies, that can be used to determine the traffic that can
+pass through the interface.
+
+**Performance: latency**
+
+Because of the traffic staying within Amazon’s networks, the physical distance travelled, the number of hops
+and the risk of traversing congested networks are all significantly lower.{' '}
+
+**Performance: bandwidth**
+
+PrivateLink supports a sustained bandwidth of 10 Gbps per availability zone with bursts up to 40 Gbps.
+
+**Performance: stability**
+
+PrivateLink consistently lowers the amount of errors and timeouts on high loads
+
+**Do you want to know if you have it properly configured? Please use this query to check it:**
```

Comment:
the following query will check for active S3 VPC Interface Endpoints?

Reply:
Agree with @aguillenv 's rephrasing.

```suggestion
**Do you want to know if you have it properly configured? The followign query will check for active S3 VPC interface endpoints:**
```